src/plugins/lua/elastic.lua

```diff
@@ -608,25 +564,34 @@ local function get_general_metadata(task)
 
   local function process_header(name)
     local hdr = task:get_header_full(name)
-    local headers_text_ignore_above = settings['index_template']['headers_text_ignore_above'] - 3
     if hdr and #hdr > 0 then
       local l = {}
       for _, h in ipairs(hdr) do
-        if settings['index_template']['headers_count_ignore_above'] ~= 0 and
-            #l >= settings['index_template']['headers_count_ignore_above']
+        if settings['index_template']['headers_count_ignore_above'] > 0 and
+          #l >= settings['index_template']['headers_count_ignore_above']
         then
           table.insert(l, 'ignored above...')
           break
         end
         local header
-        if settings['index_template']['headers_text_ignore_above'] ~= 0 and
-            h.decoded and #h.decoded >= headers_text_ignore_above
-        then
-          header = h.decoded:sub(1, headers_text_ignore_above) .. '...'
-        elseif h.decoded and #h.decoded > 0 then
-          header = h.decoded
+        local header_len
+        if h.decoded then
+          header = rspamd_text.fromstring(h.decoded)
+          header_len = header:len_utf8()
         else
-          header = empty
+          table.insert(l, empty)
+          break
+        end
+        if not header_len or header_len == 0 then
+          table.insert(l, empty)
+          break
+        end
+        if settings['index_template']['headers_text_ignore_above'] > 0 and
+          header_len >= settings['index_template']['headers_text_ignore_above']
+        then
+          header = header:sub_utf8(1, settings['index_template']['headers_text_ignore_above'])
```

Comment:
We don't have this method in LuaJIT; maybe we should be shipping something like that.

Reply:
Yep, it was added due to such need :)

Reply:
Beware that it's a method of `rspamd_text` not Lua strings.

Reply:
@vstakhov yes I understand that, but as far I tested on my local development - all data accepted just fine. First time I was do `table.insert(l, header:str)`, but just of curiosity tested without convertion to `str` and it just worked, so I left it as is to economy resources if it works. Or it could have some hidden rocks that I'm not aware off?

Reply:
btw - digest is very useful thing on a scale, allows to find unknown yet spam by just doing TOP 100 and some conditions like score < 10 😉